src/event/log_event.rs

```diff
@@ -12,43 +14,60 @@ use std::{
     iter::FromIterator,
 };
 
-#[derive(Clone, Debug, Default, Getters, PartialEq)]
+#[derive(Clone, Debug, Getters, PartialEq, Derivative, Deserialize)]
 pub struct LogEvent {
-    fields: BTreeMap<String, Value>,
+    // **IMPORTANT:** Due to numerous legacy reasons this **must** be a Map variant.
+    #[derivative(Default(value = "Value::from(BTreeMap::default())"))]
+    #[serde(flatten)]
+    fields: Value,
+
     #[getset(get = "pub")]
+    #[serde(skip)]
     metadata: EventMetadata,
 }
 
+impl Default for LogEvent {
+    fn default() -> Self {
+        Self {
+            fields: Value::Map(BTreeMap::new()),
+            metadata: EventMetadata,
+        }
+    }
+}
+
 impl LogEvent {
     pub fn new_with_metadata(metadata: EventMetadata) -> Self {
         Self {
-            fields: Default::default(),
+            fields: Value::Map(Default::default()),
             metadata,
         }
     }
 
     pub fn into_parts(self) -> (BTreeMap<String, Value>, EventMetadata) {
-        (self.fields, self.metadata)
+        (
+            self.fields.into_map().expect("fields must be a map"),
```

Comment:
Would it make sense to use `unreachable!` here too?
```suggestion
            self.fields.into_map().unwrap_or_else(|| unreachable!(fields must be a map")),
```